app/controllers/v0/sessions_controller.rb

```diff
@@ -13,8 +13,24 @@ def show
     end
 
     def destroy
-      @session.destroy
-      head :no_content
+      logout_request = OneLogin::RubySaml::Logoutrequest.new
+      logger.info "New SP SLO for userid '#{@session.uuid}'"
+
+      if saml_settings.name_identifier_value.nil?
```

Comment:
Before you change the code, would you mind running a quick manual test with two logouts.
1. Log in with browser A
2. Log in with browser B
3. Log out with browser A
4. Print the value of `saml_settings.name_identifier_value` and make sure it isnt A's `session.uuid`.  I think we might have a thread-safety issue with the settings_service which is a singleton.

```ruby
settings = SAML::SettingsService.instance.saml_settings
```